tests/linalg_test.py

```diff
@@ -269,6 +269,29 @@ def check_left_eigenvectors(a, w, vl):
     self._CompileAndCheck(partial(jnp.linalg.eig), args_maker,
                           rtol=1e-3)
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name": "_shape={}".format(
+           jtu.format_shape_dtype_string(shape, dtype)),
+       "shape": shape, "dtype": dtype, "rng_factory": rng_factory
+       }
+      for shape in [(4, 4), (5, 5), (8, 8), (7, 6, 6)]
+      for dtype in float_types + complex_types
+      for rng_factory in [jtu.rand_default]))
+  # TODO(phawkins): enable when there is an eigendecomposition implementation
+  # for GPU/TPU.
+  @jtu.skip_on_devices("gpu", "tpu")
+  def testEigvalsGrad(self, shape, dtype, rng_factory):
+    # This test sometimes fails for large matrices. I (@j-towns) suspect, but
+    # haven't checked, that might be because of perturbations causing the
+    # ordering of eigenvalues to change, which will trip up check_grads. So we
+    # just test on small-ish matrices.
+    rng = rng_factory(self.rng())
+    jtu.skip_if_unsupported_type(dtype)
+    args_maker = lambda: [rng(shape, dtype)]
+    a, = args_maker()
+    jtu.check_grads(lambda x: jnp.linalg.eigvals(x), (a,), order=1,
+                    modes=['fwd'], rtol=1e-2, atol=1e-2)
```

Comment:
Could we check `rev` mode, too? It seems like that would be a worthwhile check to include, since people definitely care about it.

Reply:
Agreed, it appears to be breaking when I turn it on, which is interesting, I'll try and get to the bottom of it tomorrow. By the way, it looks like the derivatives of `solve` might not be be covered by the tests. Searching `linalg_test.py` I've found `testSolve` methods but they don't seem to check the derivatives. Apologies if I'm missing something obvious, it's a little late here.

Reply:
To be clear, it's probably best to avoid merging until this is fixed.

Reply:
Things seem to work fine for float64 and complex128, so I'm assuming this is a numerical precision issue rather than the high level implementation being incorrect. I've enabled reverse mode tests and raised the atol and rtol for float32 and complex64 to 1e-1, I was able to lower the rtol and atol to 1e-4 for the two high precision dtypes.

Reply:
FYI I also tried constructing the random input matrix to make sure it had eigenvalues which wouldn't get re-ordered by small perturbations, but that didn't fix the failures.

Reply:
Hmm I thought that eigenvalues came out of eig in some kind of sorted order, but it seems I was wrong. So maybe there's no way to prevent small perturbations from occasionally changing the order... 😟